### PR TITLE
Discover Bugfixes Round 1

### PR DESCRIPTION
**Pull request by cuahsimarko**
Some styling and bugfixes for Discover. Functionality review can be performed at https://dev-hs-6.cuahsi.org/

Fixes https://github.com/hydroshare/hydroshare/issues/4072
Fixes https://github.com/hydroshare/hydroshare/issues/3920
Fixes https://github.com/hydroshare/hydroshare/issues/4029
Fixes https://github.com/hydroshare/hydroshare/issues/4058
Fixes https://github.com/hydroshare/hydroshare/issues/3925
Fixes https://github.com/hydroshare/hydroshare/issues/3919
Fixes https://github.com/hydroshare/hydroshare/issues/3918
Fixes https://github.com/hydroshare/hydroshare/issues/4034

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4068](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4068)

**Comment by cuahsimarko** [comment]
@sblack-usu Martin has completed the functional review could you give it a quick code review/approval?

File: hs_discover/public/index.html [hunk 1]
```diff
@@ -71,9 +71,8 @@ <h2 class="page-title">Discover
                 <div class="panel panel-default">
                     <div class="panel-heading" role="tab" id="headingLegend">
                         <h4 class="panel-title">
-                            <span class="glyphicon glyphicon-info-sign"></span>&nbsp;
-                            <a role="button" data-toggle="collapse" data-parent="#accordion" href="#legend-collapse"
-                               aria-expanded="true" aria-controls="legend-collapse">Legend</a>
+                            <a role="button" style="text-decoration: none" data-toggle="collapse" data-parent="#accordion" href="#legend-collapse"
+                               aria-expanded="true" aria-controls="legend-collapse"><span class="glyphicon glyphicon-info-sign"></span> Legend</a>
                         </h4>
```
Comment (cuahsimarko) [inline hs_discover/public/index.html:76]: include the glyphicon in the clickable anchor

File: hs_discover/public/index.html [hunk 2]
```diff
@@ -127,11 +126,6 @@ <h4 class="panel-title">
                                             <small>Public</small>
                                         </div>
 
-                                        <div class="col-sm-6">
-                                            <img src="{{ STATIC_URL }}img/private.png" alt="Private Resource"/>
-                                            <small>Private</small>
-                                        </div>
-
                                         <div class="col-sm-6">
```
Comment (cuahsimarko) [inline hs_discover/public/index.html:129]: Private resources aren't displayed in Discover so shouldn't be part of the legend

File: hs_discover/src/components/Resources.vue [hunk 3]
```diff
@@ -170,7 +170,8 @@
                                     v-for="(availability) in orderedFilter(countAvailabilities)"
                                     v-bind:key="availability">
                                     <span class="badge">{{availability[1]}}</span>
-                                    <label class="checkbox noselect" :for="'avail-'+availability[0]">{{availability[0]}}
+                                    <label class="checkbox noselect" :for="'avail-'+availability[0]">
+                                      {{availability[0].charAt(0).toUpperCase()+availability[0].slice(1)}}
                                         <input type="checkbox" class="faceted-selections" :value=availability[0]
```
Comment (cuahsimarko) [inline hs_discover/src/components/Resources.vue:175]: Title case for words Discoverable, Public, Published

File: hs_discover/src/components/Resources.vue [hunk 4]
```diff
@@ -452,6 +453,7 @@ export default {
           this.searchClick(false, false, false);
           this.setAllMarkers();
         } else {
+          this.setAllMarkers();
           recenterMap(); // eslint-disable-line
```
Comment (cuahsimarko) [inline hs_discover/src/components/Resources.vue:457]: Correct a workflow, whereby you enter maps, leave and come back after creating filters with no results. Adding this call re-renders markers (in this case having the effect of removing/clearing stale markers)

File: hs_discover/src/components/Resources.vue [hunk 5]
```diff
@@ -191,7 +192,7 @@
                 <table id="items-discovered" v-if="resources.length"
                     class="table-hover table-striped resource-custom-table main-table">
                     <thead>
-                        <tr>
+                        <tr><th><!-- placeholder --></th>
                             <th v-for="key in labels" v-bind:key="key" style="cursor:pointer"
```
Comment (cuahsimarko) [inline hs_discover/src/components/Resources.vue:196]: Remove this column header

File: hs_discover/src/components/Search.vue [hunk 6]
```diff
@@ -14,8 +14,8 @@ export default {
   name: 'Search',
   data() {
     return {
-      gridColumns: ['type', 'name', 'author', 'created', 'modified'],
-      gridColumnLabels: ['Type', 'Title', 'First Author', 'Date Created', 'Last Modified'],
+      gridColumns: ['name', 'author', 'created', 'modified'],
+      gridColumnLabels: ['Title', 'First Author', 'Date Created', 'Last Modified'],
     };
```
Comment (cuahsimarko) [inline hs_discover/src/components/Search.vue:19]: Type column header removed from table view

File: hs_discover/views.py [hunk 7]
```diff
@@ -97,9 +97,9 @@ def get(self, request, *args, **kwargs):
             if filters.get('type'):
                 for k, restype in enumerate(filters['type']):
                     if k == 0 or k == len(filters['type']):
-                        sqs = sqs.filter(resource_type_exact=Exact(restype))
+                        sqs = sqs.filter(content_type_exact=Exact(restype))
                     else:
-                        sqs = sqs.filter_or(resource_type_exact=Exact(restype))
+                        sqs = sqs.filter_or(content_type_exact=Exact(restype))
             if filters.get('availability'):
```
Comment (cuahsimarko) [inline hs_discover/views.py:103]: Use Resource Type instead of Content Type in Discover facet checkboxes

File: theme/templates/newbase.html [hunk 8]
```diff
@@ -81,7 +81,6 @@
             position: absolute;
         }
         .modal-header {
-            {#reason#}
             display: block;
```
Comment (cuahsimarko) [inline theme/templates/newbase.html:84]: Clean up old code

File: theme/templates/newbase.html [hunk 9]
```diff
@@ -42,6 +42,9 @@
         .dropdown-toggle:after {
             display: none;
         }
+        .navbar-collapse {
+            background: white;
+        }
         .navbar-fixed-top {
```
Comment (cuahsimarko) [inline theme/templates/newbase.html:48]: CSS, includes, extends and other template imports need to be thoroughly cleaned up across hydroshare. For now, a workaround overriding whatever conflict or unintended styling has cascaded to this point.

File: hs_discover/public/index.html (hunk 1 above)
Comment (hyi) [reply hs_discover/public/index.html:76]: It'd be good to put ```text-decoration: none``` into css rather than using style here in the template.

**Review by hyi — COMMENTED** [review]
@cuahsimarko I am done with code review and left one minor comment for your consideration.

File: hs_discover/public/index.html (hunk 1 above)
Comment (cuahsimarko) [reply hs_discover/public/index.html:76]: The CSS includes have a lot of sprawl and some conflict. I'd like to leave this for now, as it's very clear what is being applied and after the includes are moved and the scoped CSS is applied, where relevant the dust will settle on these types of inline styling.

**Review by hyi — APPROVED** [review]

**Pull request merged**